### ISSUE TITLE
pylint --rcfile=<file> outputting -h on success

### ISSUE THREAD

**Issue by idflores** [PyCQA/pylint#1519]
Hello! Recently found this project, and I love it! But, I ran into an issue while using it...

### The Problem
While specifying my own Pylint configuration file, running the `pylint --rcfile=<file_name>` command resulted in the same output you would get running `pylint -h` or just `pylint`. Very confusing. After some testing, I was able to confirm that the `pylint --rcfile=<file_name>` did run successfully, and putting something bogus like `pylint --rcfile` generates the proper error message, specifically "Option rcfile expects a value" and "No config file found, using default configuration".

### Steps to reproduce

This is exactly what i did from my `~` directory:

1. generate the rcfile to my profile using Terminal on MacOS
```bash
pylint --generate-rcfile > .pylintrc
```
2.  make necessary modification to the file
```bash
atom .pylintrc
```
3. switch to using new configuration file
```bash
pylint --rcfile=.pylintrc
```

### Current behavior
The same output you get when running:
```bash
pylint -h
```
VERY confusing. The output made be believe that the command was invalid. I was able to check and verify the command's success by performing another `pylint --generate-rcfile` and checking to see if my changes where included.

### Expected behavior
Some success message or nothing at all.

### pylint --version output
pylint 1.7.1,
astroid 1.5.2
Python 2.7.10 (default, Feb  7 2017, 00:08:15)
[GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.34)]

**Comment by rogalski** [comment]
I'm not really sure if there is an issue at all. Loading --rcfile without providing module to check does not make much sense.

But maybe we should explicitly mention currently used rcfile in output? So either we would have
> No config file found, using default configuration
Or
> Using config file /abs/path/to/pylintrc

This may clarify confusions like https://github.com/PyCQA/pylint/issues/1388#issuecomment-296281593

**Comment by idflores** [comment]
Thanks for taking a look @rogalski !

Just making sure we understand, the error messages like,

> No config file found, using default configuration

are showing in my experience. The issue, here, is when the the `pylintrc` file has been **successfully** set. 

#### These are an *unsuccessful* commands:

<img width="922" alt="screen shot 2017-06-02 at 12 04 02 pm" src="https://cloud.githubusercontent.com/assets/12431665/26736426/96a42c00-478b-11e7-8bda-b2e8fd93505a.png">

**Pylint Message:**
> Option rcfile expects a value

<img width="922" alt="screen shot 2017-06-02 at 12 05 07 pm" src="https://cloud.githubusercontent.com/assets/12431665/26736458/bbddd520-478b-11e7-856b-c4206e69ca5c.png">

**Pylint Message:**
> No config file found, using default configuration

<br>

#### This is a *successful* command (the confusion):

<img width="922" alt="screen shot 2017-06-02 at 12 07 41 pm" src="https://cloud.githubusercontent.com/assets/12431665/26736523/192f26b6-478c-11e7-8bd7-ff9794b5aea6.png">

Notice how this outputs the help pages again... *very* confusing in my opinion... again, I tested it, and the command did run successfully.

Here, I completely agree with your latter suggestion of adding a confirmation response:

> Using config file /abs/path/to/pylintrc

*Note: I am using iTerm2, but with minimal configuration, so no issues there...*
